### PR TITLE
Improved Lifecycle management

### PR DESCRIPTION
**Pull request by conker84**
This is a WIP PR as follow up with the eng team about how improve the lifecycle management of the Streams Plugin.
I'll provide a full description once is ready for the review.

### CODE REVIEW

File: doc/asciidoc/consumer/configuration.adoc [hunk 1]
```diff
@@ -21,6 +21,7 @@ kafka.value.deserializer=org.apache.kafka.common.serialization.ByteArrayDeserial
 
 streams.check.apoc.timeout=<ms to await for APOC being loaded, default -1 skip the wait>
 streams.check.apoc.interval=<ms interval awaiting for APOC being loaded, default 1000>
+streams.cluster.only=<boolean value that prevent for starting in single instances usefull when you're performing backup/restore ops, default false>
```
Comment (moxious) [inline doc/asciidoc/consumer/configuration.adoc:24]: Add:

"If set to true, streams will refuse to start in single instance mode, or when run in the context of the backup operation.  This is an important safety guard to ensure that operations do not occur in unexpected situations for production deploys"

File: consumer/src/main/kotlin/streams/StreamsEventSinkAvailabilityListener.kt [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                                streamsTopicService,
+                                log,
+                                db)
+            }
+        }
+        try {
+            log.info("Initialising the Streams Sink module")
+
+            // start the Sink
+            if (Neo4jUtils.isCluster(db)) {
```
[the remark this answers is not in the document]
Comment (conker84) [reply consumer/src/main/kotlin/streams/StreamsEventSinkAvailabilityListener.kt:58]: This is just to understand if we are in a cluster env by checking if the cluster procedures are present. I can change it to `dbms.cluster.overview` and add the result as debug log

File: consumer/src/main/kotlin/streams/StreamsEventSinkAvailabilityListener.kt [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                            |Cannot init the Streams Sink module as is forced to work only in a cluster env, 
+                            |please check the value of `streams.${StreamsSinkConfigurationConstants.CLUSTER_ONLY}`
+                        """.trimMargin())
+                    } else {
+                        initSinkModule()
+                    }
+                }
+            }
+
+            // Register required services for the Procedures
```
[the remark this answers is not in the document]
Comment (conker84) [reply consumer/src/main/kotlin/streams/StreamsEventSinkAvailabilityListener.kt:75]: That procedure actually does not sink you can simply return the data in order to debug it. IMHO we should not change that

**Comment by conker84** [comment]
The build is red because we added some enterprise tests that are not public available. I tested it locally and all tests are passed

File: consumer/src/main/kotlin/streams/StreamsEventConsumer.kt [hunk 4]
```diff
@@ -5,7 +5,7 @@ import streams.service.StreamsSinkEntity
 import streams.service.errors.ErrorService
 
 
-abstract class StreamsEventConsumer(private val log: Log, private val dlqService: ErrorService) {
```
[the remark this answers is not in the document]
Comment (conker84) [reply consumer/src/main/kotlin/streams/StreamsEventConsumer.kt:8]: Its lifecycle is managed totally into the implementation

**Comment by conker84** [comment]
Ok I added a maven profile that is activated only in Travis and it excludes the enterprise deps and test.

On TeamCity it will provide full tests

**Pull request merged**